ckanext/iati_generator/actions/iati.py

```diff
@@ -314,12 +320,19 @@ def iati_resources_list(context, data_dict=None):
     data_dict = data_dict or {}
     toolkit.check_access("iati_file_list", context, data_dict)
 
+    namespace_filter = data_dict.get("namespace")
+
     files_by_resource = h.iati_files_by_resource()
 
     results = []
     datasets = {}
 
     for resource_id, iati_file in files_by_resource.items():
+
+        # namespace filter
+        if namespace_filter and iati_file.namespace != namespace_filter:
+            continue
+
```

Comment:
```suggestion
```
Follow previous suggestion, implement the filter in the `iati_files_by_resource` function